python/pyiceberg/table/__init__.py

```diff
@@ -69,21 +72,288 @@
+    action = TableUpdateAction.set_default_spec
+    spec_id: int = Field(
+        alias="spec-id", description="Partition spec ID to set as the default, or -1 to set last added spec", default=-1
+    )
+
+
+class AddSortOrderUpdate(TableUpdate):
+    action = TableUpdateAction.add_sort_order
+    sort_order: SortOrder = Field(alias="sort-order")
+
+
+class SetDefaultSortOrderUpdate(TableUpdate):
+    action = TableUpdateAction.set_default_sort_order
+    sort_order_id: int = Field(
+        alias="sort-order-id", description="Sort order ID to set as the default, or -1 to set last added sort order", default=-1
+    )
+
+
+class AddSnapshotUpdate(TableUpdate):
+    action = TableUpdateAction.add_snapshot
+    snapshot: Snapshot
+
+
+class SetSnapshotRefUpdate(TableUpdate):
+    action = TableUpdateAction.set_snapshot_ref
+    ref_name: str = Field(alias="ref-name")
+    type: Literal["tag", "branch"]
+    snapshot_id: int = Field(alias="snapshot-id")
+    max_age_ref_ms: int = Field(alias="max-ref-age-ms")
+    max_snapshot_age_ms: int = Field(alias="max-snapshot-age-ms")
+    min_snapshots_to_keep: int = Field(alias="min-snapshots-to-keep")
+
+
+class RemoveSnapshotsUpdate(TableUpdate):
+    action = TableUpdateAction.remove_snapshots
+    snapshot_ids: List[int] = Field(alias="snapshot-ids")
+
+
+class RemoveSnapshotRefUpdate(TableUpdate):
+    action = TableUpdateAction.remove_snapshot_ref
+    ref_name: str = Field(alias="ref-name")
+
+
+class SetLocationUpdate(TableUpdate):
+    action = TableUpdateAction.set_location
+    location: str
+
+
+class SetPropertiesUpdate(TableUpdate):
+    action = TableUpdateAction.set_properties
+    updates: Dict[str, str]
+
+
+class RemovePropertiesUpdate(TableUpdate):
+    action = TableUpdateAction.remove_properties
+    removals: List[str]
+
+
+class Requirement(Enum):
+    assert_create = "assert-create"
+    assert_table_uuid = "assert-table-uuid"
+    assert_ref_snapshot_id = "assert-ref-snapshot-id"
+    assert_last_assigned_field_id = "assert-last-assigned-field-id"
+    assert_current_schema_id = "assert-current-schema-id"
+    assert_last_assigned_partition_id = "assert-last-assigned-partition-id"
+    assert_default_spec_id = "assert-default-spec-id"
+    assert_default_sort_order_id = "assert-default-sort-order-id"
+
+
+class TableRequirement(IcebergBaseModel):
+    requirement: Requirement
+    ref: Optional[str] = None
+    uuid: Optional[str] = None
+    snapshot_id: Optional[int] = Field(None, alias="snapshot-id")
+    last_assigned_field_id: Optional[int] = Field(None, alias="last-assigned-field-id")
+    current_schema_id: Optional[int] = Field(None, alias="current-schema-id")
+    last_assigned_partition_id: Optional[int] = Field(None, alias="last-assigned-partition-id")
+    default_spec_id: Optional[int] = Field(None, alias="default-spec-id")
+    default_sort_order_id: Optional[int] = Field(None, alias="default-sort-order-id")
+
+
+class CommitTableRequest(IcebergBaseModel):
```

Comment:
This is getting an optional `identifier` in Eduard's PR.

Reply:
Thanks! https://github.com/apache/iceberg/commit/71711e18b6a2a01da9c2ca1e7a8b4daa903399c5#diff-02549ca620d020dc9ead80088cc14e311e12a69651fa8d394cd41a4308debb2e